stl/inc/format

```diff
@@ -1496,6 +1496,214 @@ public:
+        return _Capacity;
+    }
+
+    void clear() noexcept {
+        _Size = 0;
+    }
+
+    void try_resize(const size_t _Count) {
+        try_reserve(_Count);
+        _Size = _Count <= _Capacity ? _Count : _Capacity;
+    }
+
+    void try_reserve(const size_t _New_capacity) {
+        if (_New_capacity > _Capacity) {
+            _Grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(ptrdiff_t) {}
+
+    _NODISCARD size_t count() const noexcept {
+        return 0;
+    }
+
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count = 0;
+    ptrdiff_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const ptrdiff_t _Lim) noexcept : _Limit(_Lim) {}
+
+    _NODISCARD size_t count() const noexcept {
+        return static_cast<size_t>(_Count);
+    }
+
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        size_t _Avail = static_cast<size_t>(_Limit > _Count ? _Limit - _Count : 0);
+        _Count += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Out;
+    _Ty _Data[_Fmt_buffer_size];
+
+protected:
+    void _Grow(size_t) final {
+        if (this->size() == _Fmt_buffer_size) {
+            _Flush();
+        }
+    }
+
+    void _Flush() {
+        auto _Size = this->size();
+        this->clear();
+        _Out = _STD _Copy_unchecked(_Data, _Data + this->limit(_Size), _STD move(_Out));
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _Output, ptrdiff_t _Size = _Fmt_buffer_size)
+        : _Traits(_Size), _Fmt_buffer<_Ty>(_Data, 0, _Fmt_buffer_size), _Out(_STD move(_Output)) {}
+
+    ~_Fmt_iterator_buffer() {
+        if (this->size() != 0) {
+            _Flush();
+        }
+    }
+
+    _NODISCARD _OutputIt out() {
+        _Flush();
+        return _STD move(_Out);
+    }
+
+    _NODISCARD ptrdiff_t count() const noexcept {
+        return static_cast<ptrdiff_t>(_Traits::count() + this->size());
+    }
+};
+
+template <class _Ty>
+class _Fmt_iterator_buffer<_Ty*, _Ty> final : public _Fmt_buffer<_Ty> {
+protected:
+    void _Grow(size_t) final {}
+
+public:
+    explicit _Fmt_iterator_buffer(_Ty* _Out, ptrdiff_t = 0) : _Fmt_buffer<_Ty>(_Out, 0, ~size_t()) {}
+
+    _NODISCARD _Ty* out() noexcept {
+        return this->end();
+    }
+};
+
+template <class _Container>
+class _Fmt_iterator_buffer<_STD back_insert_iterator<_Container>,
+    enable_if_t<_STD contiguous_iterator<typename _Container::iterator>, typename _Container::value_type>>

```

Comment:
Should this use `_RANGES contiguous_range` to make sure it properly handles user defined containers?

Reply:
As far as I can say we tried to keep concept support out of format

Reply:
Well, `contiguous_iterator` is already used here...

Reply:
https://github.com/microsoft/STL/blob/43d081bd05a8717b9f07bd3de2632a52e1ef6cf4/stl/inc/format#L42-L44

Reply:
Removed unnecessary `_STD`.

> Should this use _RANGES contiguous_range to make sure it properly handles user defined containers?

As commented elsewhere, it is unnecessary since user-defined containers are already supported via general specialization. If anyone wants to optimize esoteric containers, I'm sure PRs with tests would be welcome.